compiler/rustc_interface/src/util.rs

```diff
@@ -125,6 +125,7 @@ pub fn scoped_thread<F: FnOnce() -> R + Send, R: Send>(cfg: thread::Builder, f:
     let result_ptr = Ptr(&mut result as *mut _ as *mut ());
 
     let thread = cfg.spawn(move || {
+        let _ = (&run, &result_ptr);
```

Comment:
These are necessary since raw pointers aren't send/sync, but this whole block is likely to go away shortly anyway - https://github.com/rust-lang/rust/pull/89104